pkg/kubelet/pod_workers.go

```diff
@@ -1066,6 +1086,33 @@ func (p *podWorkers) completeTerminated(pod *v1.Pod) {
 	}
 }
 
+// completeUnstartedTerminated is invoked if a pod that has never been started receives a termination
+// signal before it can be started.
+func (p *podWorkers) completeUnstartedTerminated(pod *v1.Pod) {
+	p.podLock.Lock()
+	defer p.podLock.Unlock()
+
+	klog.V(4).InfoS("Pod never started and the worker can now stop", "pod", klog.KObj(pod), "podUID", pod.UID)
```

Comment:
Do we want this to be hidden at V=4? I feel like this might be worth logging at V=2 since it's unlikely to be a common thing.

Reply:
Hrm.  If all the others weren't 4 i would agree.  I think it might be useful to come back and add some of this at v3 for better insight into pod lifecycle in logs on later review.